src/raw/mod.rs

```diff
@@ -422,13 +416,19 @@ impl<T> RawTable<T> {
         dealloc(self.ctrl.as_ptr(), layout);
     }
 
+    // Pointer to the array of buckets
+    fn data(&self) -> NonNull<T> {
+        let (_, data_offset) = calculate_layout::<T>(self.buckets()).unwrap();
```

Comment:
Can you try replacing `unwrap()` with `unwrap_or_else(|| hint::unreachable_unchecked())`?

Reply:
Try calculating `data_offset` manually by copying the code from `calculate_layout`.

Reply:
Also you can make this function return a `*mut T` instead of a `NonNull<T>`.